lib/init.lua

```diff
@@ -686,12 +702,13 @@ function t.tuple(...)
 	local checks = {...}
 	return function(...)
 		local args = {...}
-		for i = 1, #checks do
-			local success, errMsg = checks[i](args[i])
+		for i, check in ipairs(checks) do
+			local success, errMsg = check(args[i])
```

Comment:
While we are on the subject of micro-optimizations, `args` could be eliminated in favor of just using `(select(i, ...))`